packages/plugin-fontawesome/src/FontawesomeNode.ts

```diff
@@ -1,8 +1,53 @@
 import { InlineNode } from '../../plugin-inline/src/InlineNode';
+import { RelativePosition } from '../../core/src/VNodes/VNode';
+import { FontawesomeDomParser } from './FontawesomeDomParser';
 
 export class FontawesomeNode extends InlineNode {
     static readonly atomic = true;
     constructor(public htmlTag = 'SPAN') {
         super();
     }
+
+    //--------------------------------------------------------------------------
+    // Lifecycle
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    locate(domNode: Node, offset: number): [this, RelativePosition] {
+        if (domNode.nodeType === Node.TEXT_NODE) {
+            // We are targetting one of the invisible characters surrounding the
+            // fontawesome node.
+            if (FontawesomeDomParser.isFontawesome(domNode.previousSibling)) {
+                // We are targetting the invisible character AFTER the
+                // fontawesome node.
+                if (offset === 0) {
+                    // Moving from after the fontawesome node to before it.
+                    // (DOM is `<invisible/><fontawesome/>[]<invisible/>` but
+                    // should be `[]<invisible/><fontawesome/><invisible/>`).
+                    return [this, RelativePosition.BEFORE];
```

Comment:
We need to move this `locate` stuff. It is DOM stuff so it should be on the renderer. If it was ideed on the renderer, you could simply read the `engine.renderings` object rather than having to "observe" your DOM surroundings. Something like this:
```typescript
class FontAwesomeDomRenderer {
    locate(domNode: Node, offset: number, node: VNode): [this, RelativePosition] {
        const rendering = this.engine.renderings.get(node);
        if (domNode === rendering[rendering.length - 1]) {
            // We are targetting the invisible character AFTER the
                // fontawesome node.
            if (offset === 0) {
                // Moving from after the fontawesome node to before it.
                // (DOM is `<invisible/><fontawesome/>[]<invisible/>` but
                // should be `[]<invisible/><fontawesome/><invisible/>`).
                return [this, RelativePosition.BEFORE];
            } else {
                // Stay after the fontawesome node.
                return [this, RelativePosition.AFTER];
            }
        } else if (domNode === rendering[0]) {
            // We are targetting the invisible character BEFORE the
            // fontawesome node.
            if (offset === 1) {
                 // Moving from before the fontawesome node to after it.
                // (DOM is `<invisible/>[]<fontawesome/><invisible/>` but
                // should be `<invisible/><fontawesome/><invisible/>[]`).
                return [this, RelativePosition.AFTER];
            } else {
                // Stay before the fontawesome node.
                return [this, RelativePosition.BEFORE];
            }
        }  else {
            // We should never reach this. It's a fallback to prevent a crash in
            // the eventuality that it would.
            return [this, RelativePosition.BEFORE];
        }
    }
}
```

In particular, your code will fail in the event that some node that is not a `FontAwesomeNode` ends up rendering itself using a node that your regex would recognize as being a fontawesome. If it does, the first invisible character would be mistaken for the second one because its "previousSibling" would be recognized as a fontawesome even though it is not the one that this invisible character is "attached" to. Hence my suggestion of using the `renderings` object to specifically work on the nodes rendered by that VNode and only that.

Reply:
Right but if I do that I have to move `locate` for everything else, don't I? I think it's a little bit beyond the scope of this PR.

Reply:
Well yes you'd have to move `locate` for everything else. It's out of the scode of this PR but it will have to be done eventually, as the code is soemwhat wrong here.

Reply:
OK I'm opening an issue for it.